src/nanoemoji/svg.py

```diff
@@ -23,7 +23,17 @@
 from picosvg import svg_meta
 from picosvg.svg_transform import Affine2D
 import regex
-from typing import MutableMapping, Sequence, Tuple
+from typing import MutableMapping, NamedTuple, Sequence, Tuple
+
+
+class ReuseCache(NamedTuple):
+    old_to_new_id: MutableMapping[str, str]
+    gradient_to_id: MutableMapping[str, str]
+    shapes: MutableMapping[str, etree.Element]
+
+    @classmethod
+    def create(cls):
```

Comment:
the fact that this contains _mutable_ dicts makes this an awkward tuple. How about we turn it into a dataclass, and do `old_to_new_id: Dict[str, str] = dataclasses.field(default_factory=dict)`, etc. to address the mutability issue of the default `{}` values?

Reply:
I think my theory was I didn't want the actual dict to be replaced. In looking at it again +1 to dataclasses. I left the type as MutableMapping.